dali/operators/generic/constant.h

```diff
@@ -80,9 +80,9 @@ class Constant : public Operator<Backend> {
 
   bool SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) override {
     output_desc.resize(1);
-    if (output_shape_.empty()) {
-      int batch_size = this->spec_.template GetArgument<int>("batch_size");
-      output_shape_ = uniform_list_shape(batch_size, shape_arg_);
+    auto curr_batch_size = ws.GetRequestedBatchSize(0);
+    if (output_shape_.empty() || output_shape_ .num_elements() < curr_batch_size) {
```

Comment:
```suggestion
    if (output_shape_.empty() || output_shape_ .num_elements() != curr_batch_size) {
```
Shouldn't this happen on change?

Reply:
Done